trio/_core/_io_windows.py

```diff
@@ -530,6 +531,12 @@ def process_events(self, received):
             elif entry.lpCompletionKey == CKeys.WAIT_OVERLAPPED:
                 # Regular I/O event, dispatch on lpOverlapped
                 waiter = self._overlapped_waiters.pop(entry.lpOverlapped)
+                overlapped = entry.lpOverlapped
+                transferred = entry.dwNumberOfBytesTransferred
+                info = CompletionKeyEventInfo(
+                    lpOverlapped=overlapped, dwNumberOfBytesTransferred=transferred
+                )
+                self._overlapped_entries[overlapped] = info
```

Comment:
You can pass the structure into the task as you reschedule it: `_core.reschedule(waiter, Value(info))` (`from outcome import Value`). Then you don't need the sideband `_overlapped_entries` dict.

Reply:
Took awhile to dig into `wait_task_rescheduled` to grok what was going on there, but that's way easier than what I had originally done, thanks!